src/test/java/org/jabref/gui/ScreenshotDarkTheme.java

```diff
@@ -0,0 +1,78 @@
+package org.jabref.gui;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collections;
+
+import javafx.stage.Stage;
+
+import org.jabref.Globals;
+import org.jabref.JabRefException;
+import org.jabref.JabRefGUI;
+import org.jabref.JabRefMain;
+import org.jabref.gui.util.ThemeLoader;
+import org.jabref.preferences.JabRefPreferences;
+import org.jabref.testutils.category.GUITest;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testfx.api.FxRobot;
+import org.testfx.framework.junit5.ApplicationExtension;
+import org.testfx.framework.junit5.Start;
+import org.testfx.framework.junit5.Stop;
+import org.testfx.util.DebugUtils;
+
+/**
+ * JabRef requires a restart to apply the dark theme. Thus, screenshotting is is implemented in a separate class
+ */
+@GUITest
+@ExtendWith(ApplicationExtension.class)
+class ScreenshotDarkTheme {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ScreenshotDarkTheme.class);
+    private static final Path ROOT = Path.of("build/screenshots").toAbsolutePath();
+
+    @BeforeAll
+    static void createDirectory() throws IOException {
+        if (!Files.exists(ROOT)) {
+            Files.createDirectory(ROOT);
+        }
+    }
+
+    @Start
+    public void onStart(Stage stage) throws JabRefException {
+        final JabRefPreferences preferences = JabRefPreferences.getInstance();
+        Globals.prefs = preferences;
```

Comment:
Is it really necessary to overwrite the `Globals.prefs` ?

The current code also writes to the active preferences, which means that normal JabRef installations are affected as well.

Reply:
Overwriting: This basically is https://github.com/JabRef/jabref/blob/43f533bcf6e7729925e21b58f30c48becc584f30/src/main/java/org/jabref/JabRefMain.java#L56. - I tried to start a full JabRef as the screenshots will show a complete JabRef and not some mocked part.

Next step is to mock the preferences. Did not find out to do it in my JabRef timeslot (especially https://github.com/JabRef/jabref/blob/43f533bcf6e7729925e21b58f30c48becc584f30/src/main/java/org/jabref/preferences/JabRefPreferences.java#L438 caused me headaches. Why is it in the constructor?). (Refs https://github.com/JabRef/jabref/pull/2768.) I think, whole JabRef has to be rewritten to rely on https://github.com/JabRef/jabref/blob/43f533bcf6e7729925e21b58f30c48becc584f30/src/main/java/org/jabref/preferences/PreferencesService.java#L30. Maybe, that is too much and I just have to mock `JabRefPreferences`. 

Thus,  I went for the quick way to show an end-to-end way to get screenshots. In case I cannot do that "in time", I would suggest to split up the PR: A) work on having a clean way for the preferences B) finish the basis for others to add new screenshots.